sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -34,7 +35,8 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
   private val createCondition = lock.newCondition()
 
   @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE)
-  internal val eventQueue = LinkedList<Runnable>()
+  internal val renderEventQueue = CopyOnWriteArrayList<Runnable>()
+  private val eventQueue = CopyOnWriteArrayList<Runnable>()
```

Comment:
There are many adding operation for these two queues,  `CopyOnWriteArrayList` might get a lower performance.

Reply:
Despite we need those queues to be thread-safe, I'll rework an approach a bit.

Reply:
What is the use-case for copy on write? are we mutating these objects from multiple threads? or is it to support a removing while in-use setup?

Reply:
What is the use-case for copy on write? are we mutating these objects from multiple threads? or is it to support a removing while in-use setup?

edit: you already replied in the meantime above :) can we avoid access from multiple threads? I'm conceptually ok with using thread safe components to not push the responsibility of synchronization on us.

Reply:
Yes, I tried an approach with using mutex but it looks clumsy.